### PR TITLE
add functional tests + use savepoints if possible

### PR DESCRIPTION
**Pull request by dmaicher**
- add functional test that use mysql DB
- use savepoints if possible for nested transactions as it makes rolling back transactions easier

### CODE REVIEW

File: src/DAMA/DoctrineTestBundle/Doctrine/DBAL/StaticConnectionFactory.php [hunk 1]
```diff
@@ -46,7 +46,13 @@ public function createConnection(array $params, Configuration $config = null, Ev
 
         if (StaticDriver::isKeepStaticConnections()) {
             // The underlying connection already has a transaction started.
-            // So we start it as well on this connection so the internal state ($_transactionNestingLevel) is in sync with the underlying connection.
+            // Make sure we use savepoints to be able to easily roll-back nested transactions
+            if ($connection->getDriver()->getDatabasePlatform()->supportsSavepoints()) {
+                $connection->setNestTransactionsWithSavepoints(true);
```
Comment (dmaicher) [inline src/DAMA/DoctrineTestBundle/Doctrine/DBAL/StaticConnectionFactory.php]: @garak @Jean85 are you actively using this bundle? If so can you confirm that this does not break any of your testsuites?

I tested it with multiple of my projects including mysql, postgres and sqlite databases and for me it works fine.

Comment (Jean85) [reply src/DAMA/DoctrineTestBundle/Doctrine/DBAL/StaticConnectionFactory.php]: Yes I'm using it, but only with MySQL. I'll try it as soon as I can.

**Comment by Jean85** [comment]
Damn, since we allowed PHPUnit 6 it makes those tests fail with the legacy listener.
IMHO you should run those tests only with PHPUnit 5 under PHPUnit 5. Probably `@requires` is helpful here: https://phpunit.de/manual/current/en/incomplete-and-skipped-tests.html#incomplete-and-skipped-tests.skipping-tests-using-requires

File: src/DAMA/DoctrineTestBundle/Doctrine/DBAL/StaticConnectionFactory.php (hunk 1 above)
Comment (dmaicher) [reply src/DAMA/DoctrineTestBundle/Doctrine/DBAL/StaticConnectionFactory.php]: @Jean85 any chance to test this?

**Comment by dmaicher** [comment]
@Jean85 actually I found a nicer way so we don't even need two different classes for the listener :blush:

**Comment by Jean85** [comment]
Yep I've seen it, it's a nice way to conflate those two classes. BWT, you should either bump to a major version or implement some alias to avoid a BC for those that where using a different (`Legacy-`) listener class name.

**Comment by dmaicher** [comment]
Yes indeed makes sense to not break BC in a minor release :+1: I will just relase this as 4.0.0.

File: src/DAMA/DoctrineTestBundle/Doctrine/DBAL/StaticConnectionFactory.php (hunk 1 above)
Comment (dmaicher) [reply src/DAMA/DoctrineTestBundle/Doctrine/DBAL/StaticConnectionFactory.php]: I would like to confirm it does not break things for you before releasing this :blush:

**Pull request merged**